src/NuGet.Core/NuGet.Packaging/Signing/Utility/SigningUtility.cs

```diff
@@ -272,7 +272,7 @@ public static async Task SignAsync(SigningOptions options, SignPackageRequest si
             {
                 if (unsignedPackageStream != null && !ReferenceEquals(unsignedPackageStream, options.InputPackageStream))
                 {
-                    unsignedPackageStream.Dispose();
+                    await unsignedPackageStream.DisposeAsync();
```

Comment:
I don't believe this will compile on .NET Framework. Did you try to build your changes? If so, are you on Linux? Unfortunately, our build scripts try to be helpful by excluding .NET Framework TFMs when building on Linux and Mac.